### ISSUE TITLE
Do sound files need to be republished?

### ISSUE THREAD

**Issue by johnjung** [johnjung/lucy#4]
The first sound file in the English language column on https://lucy.lib.uchicago.edu/?uuid=0ad43a89-09d6-4292-88cb-b6fd6dfe41e5&section=1 is for a sound clip of a speaker saying "Hi there, brother!" If I click this link, I go to the following HTML page:

http://ochre.lib.uchicago.edu/ochre?uuid=12cc2b30-d304-4a40-ae14-11d766b66e2f

In the XML data, this UUID appears in the following element:

&lt;resource type="audio" uuid="12cc2b30-d304-4a40-ae14-11d766b66e2f"&gt;Lesson 1, sentence 01, English&lt;/resource&gt;

Do sound files need to be republished?

**Comment by MillerPr** [comment]
Had the sound files been working previously?

I republished the uuid you cited: 12cc... Let me know if it works now.

Hmm, what changed?

**Comment by johnjung** [comment]
Ok, I think I see what's happening. If I request http://ochre.lib.uchicago.edu/ochre?uuid=12cc2b30-d304-4a40-ae14-11d766b66e2f, I see that there is an &lt;identification&gt; element in the XML that gets returned. That element has an "iri" attribute that points to the location of the appropriate sound file on lucy.lib. Please let me know if this sounds correct:

For each sound file on a page, retrieve XML from OCHRE using something like the ochre.lib URL above. Look up the //identification/@iri for the location of the .mp3 file. 

This adds a few requests to each page with sound files, but I should be able to do them via JavaScript on the front end, so it won't slow down the page load time. Please let me know if this is the best way to get the sound files, or if there is a better way. 

Thank you!

**Comment by MillerPr** [comment]
Yes, I think what you are describing is the correct way to get the published iri on lucy.lib. This was a recent update to our strategy to allow projects to expose file locations on a case-by-case basis.

**Comment by johnjung** [comment]
Thanks very much- this sound good. I'll close this issue out.

**Comment by johnjung** [comment]
I'm reopening this issue. Looking at the XML for lesson one:
http://ochre.lib.uchicago.edu/ochre?uuid=0ad43a89-09d6-4292-88cb-b6fd6dfe41e5

To retrieve audio links, I get uuid attributes from the following nodes:
//links/resource[@type='audio']

The uuid for lesson one, sentence one is 12cc2b30-d304-4a40-ae14-11d766b66e2f. When I retrieve that XML from http://ochre.lib.uchicago.edu/ochre?uuid=12cc2b30-d304-4a40-ae14-11d766b66e2f, I see an //identification[@iri] element. 

However, the uuid for lesson one, sentence one, breakdown one is 93cc3a42-88c9-414f-b263-02c785ab479b. When I retrieve that XML from http://ochre.lib.uchicago.edu/ochre?uuid=93cc3a42-88c9-414f-b263-02c785ab479b, I do not find the an identification element with an iri attribute. 

Could you please take a look and let me know what's happening here? Should I be able to retrieve the URL for breakdown mp3s in the same way I retrieve the url for sentence mp3s?

**Comment by MillerPr** [comment]
It looks like these needed to be republished after the new IRI attribute was added. They are now republished and should have what you need. Let me know if you notice any other resources with this problem.

**Comment by johnjung** [comment]
Thanks for looking at this. Now I can find the iri attribute for "Hi there, brother!" in the first lesson, but I can't find it for other basic sentences in lesson one. For example, here is the URL for "Hola, suku'un!":

http://ochre.lib.uchicago.edu/ochre?uuid=067c3fb9-9afe-4a27-95e4-bc0d4000eb2b

The XML at this endpoint doesn't currently include the strings "iri" or "mp3".

**Comment by MillerPr** [comment]
These needed to be republished as well. Done.

Let me know if you notice others. I'll take a quick browse through the other hierarchies of audio files to see if I notice an others without iri or mp3.

**Comment by johnjung** [comment]
These look great now. I added some error handling to the JavaScript that retrieves the links to these sound files: if you have a web inspector open when you visit lucy.lib, you'll see a message "Unable to locate audio file for the following element" if any sound files can't be located for that page. I looked at a few pages and they all look good. 

I'll close this issue out now- we can reopen it if necessary.
